backend/src/nodes/nodes/pytorch/convert_to_onnx.py

```diff
@@ -12,25 +12,34 @@
 from ...utils.exec_options import get_execution_options
 from ...utils.pytorch_utils import to_pytorch_execution_options
 from ...utils.onnx_model import OnnxModel
-from ...utils.torch_types import PyTorchModel
+from ...utils.torch_types import PyTorchSRModel, isPyTorchSRModel
 
 
 @NodeFactory.register("chainner:pytorch:convert_to_onnx")
 class ConvertTorchToONNXNode(NodeBase):
     def __init__(self):
         super().__init__()
         self.description = """Convert a PyTorch model to ONNX."""
-        self.inputs = [ModelInput("PyTorch Model")]
+        self.inputs = [
+            ModelInput(
+                "PyTorch Model",
+                input_type="PyTorchModel { arch: invStrSet(PyTorchModel::FaceArchs) }",
+            )
+        ]
         self.outputs = [OnnxModelOutput(label="ONNX Model")]
 
         self.category = PyTorchCategory
         self.name = "Convert To ONNX"
         self.icon = "ONNX"
         self.sub = "Utility"
 
-    def run(self, model: PyTorchModel) -> OnnxModel:
+    def run(self, model: PyTorchSRModel) -> OnnxModel:
         exec_options = to_pytorch_execution_options(get_execution_options())
 
+        assert isPyTorchSRModel(
+            model
+        ), "Only normal SR models can be converted to ONNX at this time."
```

Comment:
From a type perspective, this doesn't make any sense. You already declared that the type of `model` is an SR torch model, so `isPyTorchSRModel` should always return true. Of course, that's not what actually happens because the type of `model` is simply wrong. `model` is actually a `PyTorchModel` (no SR).

There are 2 ways to fix this:
1. Move the runtime type into the `enforce` method of the input. `model: PyTorchSRModel` will now be correct and the assertion in this `run` method can be removed.
2. Type narrowing using [type guards](https://peps.python.org/pep-0647/). I *think* you used those wrong type annotations because python didn't understand that `isPyTorchSRModel` and `isPyTorchFaceModel` are just `isinstance` checks, and so it didn't narrow the type of `model: PyTorchModel` correctly.

    This can be fixed by using a type guard like this:
    ```py
    def isPyTorchFaceModel(model: PyTorchModel) -> TypeGuard[PyTorchFaceModel]:
        return isinstance(model, PyTorchFaceModels)
    ```

    Note that while type guards are a relatively standard feature for type checkers, it's a recent addition (py 3.10) to python, so we'll need [type-extensions](https://pypi.org/project/typing-extensions/). While this is another dep, I don't think it's a bad idea to add it. I already encountered a few situations where I needed a py 3.11 type and had to find another solution instead.

Reply:
I'll just add to the enforce method